Lib/unittest/test/testmock/testmock.py

```diff
@@ -1566,6 +1566,22 @@ def test_attribute_deletion(self):
             self.assertRaises(AttributeError, getattr, mock, 'f')
 
 
+    def test_attribute_deletion_reset_mock(self):
+        mock = Mock()
```

Comment:
I would just leave this test as:

```
mock = Mock()
mock.child = True
del mock.child
mock.reset_mock()
```

And rename it to "test_reset_mock_does_not_raise_on_attr_deletion". The rest is verified in other test cases.